Use 64-bit bit buffer when emitting codes.

nmoinvaz opened the pull request:
This PR changes the bit buffer to be 64-bit. Maybe it will reduce memory copies by half. Also `zng_emit_dist` now calls `send_bits` only once.

nmoinvaz committed:
Use 64-bit bit buffer when emitting codes.

nmoinvaz commented:
Windows GCC ci seems to keep failing intermittently due to mwing download issues..

Dead2 commented:
Yeah, I have noticed. Quite annoying, wish GA had a caching proxy for such things.

Dead2 commented:
This looks very nice, it gives us around 0.37% faster compression.
Strangely, deflate_slow again sticks out as becoming slighly slower, while all the rest become faster..
```
#HEAD
   text    data     bss     dec     hex filename
  86655    1216      32   87903   1575f libz-ng.so.1

 Runs: 30
 Levels: 1-8
 Skipworst: True (15)

 Level   Comp   Comptime min/avg/max  Decomptime min/avg/max
 1      49.603%    0.227/0.232/0.235       0.129/0.133/0.135  20804915
 2      38.062%    0.408/0.417/0.421       0.130/0.135/0.137  15964205
 3      37.021%    0.516/0.525/0.530       0.128/0.131/0.134  15527698
 4      35.832%    0.587/0.599/0.606       0.124/0.129/0.132  15028898
 5      35.719%    0.642/0.653/0.659       0.122/0.128/0.131  14981562
 6      35.651%    0.788/0.800/0.810       0.125/0.129/0.132  14953264
 7      35.030%    1.442/1.457/1.467       0.124/0.128/0.131  14692564
 8      34.995%    2.232/2.249/2.269       0.124/0.127/0.130  14677992

 avg1   37.739%                0.867                   0.130
 tot                         103.996                  15.599
```
```
# PR/579
   text    data     bss     dec     hex filename
  87127    1216      32   88375   15937 libz-ng.so.1

 Level   Comp   Comptime min/avg/max  Decomptime min/avg/max
 1      49.603%    0.217/0.222/0.224       0.127/0.133/0.136  20804915
 2      38.062%    0.408/0.414/0.418       0.132/0.136/0.138  15964205
 3      37.021%    0.512/0.519/0.524       0.130/0.133/0.135  15527698
 4      35.832%    0.583/0.591/0.596       0.125/0.129/0.132  15028898
 5      35.719%    0.635/0.647/0.656       0.125/0.128/0.131  14981562
 6      35.651%    0.778/0.795/0.803       0.125/0.129/0.132  14953264
 7      35.030%    1.451/1.468/1.483       0.125/0.128/0.130  14692564
 8      34.995%    2.237/2.253/2.272       0.124/0.128/0.130  14677992

 avg1   37.739%                0.863                   0.130
 tot                         103.610                  15.654
```

Dead2 commented:
This seems to trigger CIFuzz

nmoinvaz commented:
This should be resolved now:

I changed `s->pending + 4 >= s->pending_buf_size` to `s->pending + ((BIT_BUF_SIZE + 7) >> 3) >= s->pending_buf_size`.  (Same calculation is in _deflatePrime_) It would not flush the buffer soon enough due to increased size of bit buffer. I have changed the code so that it uses `BIT_BUF_SIZE` #define from now on.

Dead2 commented:
```
# HEAD
   text    data     bss     dec     hex filename
  90813    1280      32   92125   167dd libz-ng.so.1

 Level   Comp   Comptime min/avg/max  Decomptime min/avg/max
 1      49.603%    0.226/0.229/0.231       0.129/0.133/0.135  20804915
 2      38.062%    0.432/0.438/0.441       0.130/0.136/0.138  15964205
 3      37.021%    0.529/0.536/0.543       0.129/0.134/0.135  15527698
 4      35.832%    0.586/0.597/0.602       0.126/0.129/0.131  15028898
 5      35.719%    0.638/0.647/0.653       0.126/0.129/0.132  14981562
 6      35.651%    0.776/0.796/0.804       0.127/0.130/0.131  14953264
 7      35.030%    1.447/1.462/1.477       0.120/0.127/0.130  14692564
 8      34.995%    2.232/2.251/2.270       0.125/0.129/0.131  14677992

 avg1   37.739%                0.870                   0.131
 tot                         104.352                  15.696
```
```
# PR/579 70538c58daab7bb8af73e26af5423d546a9b1ea9
   text    data     bss     dec     hex filename
  91282    1280      32   92594   169b2 libz-ng.so.1

 Level   Comp   Comptime min/avg/max  Decomptime min/avg/max
 1      49.603%    0.221/0.226/0.228       0.129/0.133/0.135  20804915
 2      38.062%    0.429/0.433/0.436       0.130/0.134/0.136  15964205
 3      37.021%    0.524/0.534/0.541       0.122/0.130/0.133  15527698
 4      35.832%    0.583/0.594/0.599       0.126/0.129/0.131  15028898
 5      35.719%    0.641/0.649/0.654       0.126/0.129/0.132  14981562
 6      35.651%    0.781/0.794/0.803       0.124/0.129/0.132  14953264
 7      35.030%    1.434/1.458/1.475       0.119/0.126/0.129  14692564
 8      34.995%    2.224/2.248/2.264       0.124/0.128/0.130  14677992

 avg1   37.739%                0.867                   0.130
 tot
